packages/ag-charts-enterprise/src/box-plot/boxPlotSeries.ts

```diff
@@ -0,0 +1,319 @@
+                q3Value: Math.round(xAxis.scale.convert(q3Value, { strict: false })),
+                maxValue: Math.round(xAxis.scale.convert(maxValue, { strict: false })),
+            };
+
+            context.nodeData.push(nodeData);
+        });
+
+        return [context];
+    }
+
+    async processData(dataController: _ModuleSupport.DataController): Promise<void> {
+        const { yKey, minKey, q1Key, medianKey, q3Key, maxKey, data = [] } = this;
+
+        if (!yKey || !minKey || !q1Key || !medianKey || !q3Key || !maxKey) return;
+
+        const { dataModel, processedData } = await dataController.request(this.id, data, {
+            props: [
+                keyProperty(this, yKey, false, { id: `yValue` }),
+                valueProperty(this, minKey, true, { id: `minValue` }),
+                valueProperty(this, q1Key, true, { id: `q1Value` }),
+                valueProperty(this, medianKey, true, { id: `medianValue` }),
+                valueProperty(this, q3Key, true, { id: `q3Value` }),
+                valueProperty(this, maxKey, true, { id: `maxValue` }),
+            ],
+            dataVisible: this.visible,
+        });
+
+        this.dataModel = dataModel;
+        this.processedData = processedData;
+    }
+
+    // @ts-ignore
+    getDomain(direction: _ModuleSupport.ChartAxisDirection): any[] {
+        const { processedData, dataModel } = this;
+        if (!(processedData && dataModel)) return [];
+
+        if (direction === ChartAxisDirection.X) {
+            const minValues = dataModel.getDomain(this, `minValue`, 'value', processedData);
+            const maxValues = dataModel.getDomain(this, `maxValue`, 'value', processedData);
+
+            return this.fixNumericExtent(
+                [Math.min(...minValues), Math.max(...maxValues)],
+                this.axes[ChartAxisDirection.X]
+            );
+        } else {
+            const yValueIndex = dataModel.resolveProcessedDataIndexById(this, `yValue`).index;
+            return processedData.domain.keys[yValueIndex];
+        }
+    }
+
+    getLegendData(): _ModuleSupport.ChartLegendDatum[] {
+        return [];
+    }
+
+    // @ts-ignore
+    getTooltipHtml(seriesDatum: any): string {
+        return '';
+    }
+
+    protected isLabelEnabled(): boolean {
+        return false;
+    }
+
+    protected async updateDatumSelection(opts: {
+        nodeData: BoxPlotNodeDatum[];
+        datumSelection: _Scene.Selection<_Scene.Group, BoxPlotNodeDatum>;
+    }) {
+        const { nodeData, datumSelection } = opts;
+        const data = nodeData ?? [];
+        return datumSelection.update(
+            data.filter(
+                (datum) =>
+                    datum.minValue <= datum.q1Value &&
+                    datum.q1Value <= datum.medianValue &&
+                    datum.medianValue <= datum.q3Value &&
+                    datum.q3Value <= datum.maxValue
+            )
```

Comment:
I would do this filtering earlier on, in the `processData()` step (you might be able to configure a data validator for this purpose).

Reply:
I moved it to `processData`, if you think I should add a validator then let's talk about it